libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccount.m

```diff
@@ -84,7 +84,8 @@ - (instancetype)initWithCredentials:(SFOAuthCredentials*) credentials {
     self = [super init];
     if (self) {
         _observingCredentials = NO;
-        self.credentials = credentials;
+        _credentials = credentials;
+        _loginState = (credentials.refreshToken.length > 0 ? SFUserAccountLoginStateLoggedIn : SFUserAccountLoginStateNotLoggedIn);
         _syncQueue = dispatch_queue_create(kSyncQueue, NULL);
     }
```

Comment:
So if there is a refresh token we assume the user is logged in, although it may be expired. In that case the user will be logged out after attempting to use it. Makes sense.